Remove dependency on keccak NPM module.

The `keccak` NPM module is a native module which requires `node-gyp` to properly install.  This means the developer needs a fully Python dev-env setup, which can be problematic with Python 2.7 vs 3.0.  Please use a native JS sha3 library rather than keccak so users don't need to install Python just to use `solc` without error.

A good alternative library for keccak hashing is https://github.com/emn178/js-sha3.  It has no dependencies and works in node and browser.

```
PS > npm install

> keccak@1.4.0 install C:\...\node_modules\keccak
> npm run rebuild || echo "Keccak bindings compilation fail. Pure JS implementation will be used."


> keccak@1.4.0 rebuild C:\...\node_modules\keccak
> node-gyp rebuild


C:\...\node_modules\keccak>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:509:16)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\graceful-fs\polyfills.js:284:29
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:154:21)
gyp ERR! System Windows_NT 10.0.17763
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\...\node_modules\keccak
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! keccak@1.4.0 rebuild: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the keccak@1.4.0 rebuild script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\...\AppData\Local\npm\npmrc\_logs\2019-05-05T02_48_08_179Z-debug.log
```

--- a/downloadCurrentVersion.js
+++ b/downloadCurrentVersion.js
@@ -7,7 +7,7 @@ var pkg = require('./package.json');
 var fs = require('fs');
 var https = require('https');
 var MemoryStream = require('memorystream');
-var ethJSUtil = require('ethereumjs-util');
+var keccak256 = require('js-sha3').keccak256;
 
 function getVersionList (cb) {
   console.log('Retrieving available version list...');
@@ -48,7 +48,7 @@ function downloadBinary (outputName, version, expectedHash) {
     response.pipe(file);
     file.on('finish', function () {
       file.close(function () {
-        var hash = '0x' + ethJSUtil.sha3(fs.readFileSync(outputName, { encoding: 'binary' })).toString('hex');
+        var hash = '0x' + keccak256(fs.readFileSync(outputName, { encoding: 'binary' }));
         if (expectedHash !== hash) {
           console.log('Hash mismatch: ' + expectedHash + ' vs ' + hash);
           process.exit(1);

--- a/linker.js
+++ b/linker.js
@@ -1,11 +1,7 @@
-var keccak = require('keccak');
-
-function keccak256 (input) {
-  return keccak('keccak256').update(input).digest();
-}
+var keccak256 = require('js-sha3').keccak256;
 
 function libraryHashPlaceholder (input) {
-  return '$' + keccak256(input).toString('hex').slice(0, 34) + '$';
+  return '$' + keccak256(input).slice(0, 34) + '$';
 }
 
 var linkBytecode = function (bytecode, libraries) {

--- a/package.json
+++ b/package.json
@@ -43,16 +43,15 @@
   "dependencies": {
     "command-exists": "^1.2.8",
     "fs-extra": "^0.30.0",
-    "keccak": "^1.0.2",
     "memorystream": "^0.3.1",
     "require-from-string": "^2.0.0",
     "semver": "^5.5.0",
+    "js-sha3": "0.8.0",
     "tmp": "0.0.33",
     "yargs": "^11.0.0"
   },
   "devDependencies": {
     "coveralls": "^3.0.0",
-    "ethereumjs-util": "^5.2.0",
     "istanbul": "^0.4.5",
     "semistandard": "^12.0.0",
     "tape": "=4.9.2",

